packages/backend/src/managers/playgroundV2Manager.ts

```diff
@@ -87,7 +87,7 @@ export class PlaygroundV2Manager extends Publisher<PlaygroundV2[]> implements Di
    * @param userInput the user input
    * @param options the model configuration
    */
-  async submit(playgroundId: string, userInput: string, options?: ModelOptions): Promise<void> {
+  async submit(playgroundId: string, userInput: string, systemPrompt: string, options?: ModelOptions): Promise<void> {
```

Comment:
The systemPrompt is not something we can/should change once the conversation is started ? 

The conversation history should not be changed, as it would lead to incoherence if we are able to edit it while we already have existing messages.

Looking at how works ChatGPT or else, it is always defined at the begining of the conversation. In the same way that the model id is defined at the conversation creation.

The systemprompt should be provided in the creation request and added to the conversation, or it should have its own endpoint to change it, and once the conversation is started you cannot edit it. (disabling text area)

We could imagine 

````ts

setConversationSystemPrompt(conversationId: string, content: string): void {
    const conversation = this.conversations.get(conversationId);
    //ensure exists
    if(conversation .length > 1) throw new Error('cannot change on started conversation');
    // set system prompt value
}
```

Reply:
It seems to me that it is not what @slemeur defined in the spec:

```
How does it work? 
- The settings are applied to the NEXT call to the model to be made and will be sent in the payload made to the model.
- That means that, each time we do a call to the model, we read the state of the settings, get the values and append that in the API call.
```

Reply:
IMO The quoted statement does not reflect how a system prompt in LLM conversation works @slemeur

Reply:
The goal is to let the user being able to iterate with the different "settings" on how the prompt is getting configured. 
- Say the user is trying with one set of settings.
- Get the answer and want to see if adjusting the settings will make the answer any better.
Hence the need to apply the settings only to the next prompt sent to the model - so the user can keep tuning;

Reply:
I think there is a misunderstood, changing a settings should only apply to the next messages, the system prompt is something that the model is receiving at the start of the conversation.

If you change the system prompt, you are changing the first message, (system prompt is simply a message, with a role `system`.) 

In chatGPT when you change a message in a conversation, **all the messages bellow are deleted** 

https://github.com/projectatomic/ai-studio/assets/42176370/cd85fdd2-e911-42b0-aa69-a3f998c7f6b6

You cannot expect in an existing covnersation to be able to change the system prompt, as the system prompt is the first message of the conversation. Otherwise you should delete all the messages already sent. 

I am not saying that a chat bot is deterministic, but at least coherent, if you have an existing conversation, and you change the system prompt, then you re-sent a message, the existing conversation is based an a message that is not here anymore. It does not make sense.

Reply:
In this conversation, I have set the system prompt as 'reply in french' at the beginning, then changed to 'reply in italian' before sending the second prompt. The model replies as expected.

![reply-french-italian](https://github.com/projectatomic/ai-studio/assets/9973512/9e971793-c0c8-460a-8868-33762d53e61f)